rpcs3/Emu/System.cpp

```diff
@@ -2472,17 +2472,23 @@ void Emulator::GracefulShutdown(bool allow_autoexit, bool async_op, bool savesta
 	}
 	else
 	{
+		// TODO: Need to be always async
```

Comment:
why?

Reply:
Because looping on raw QApplication::processEvents() is a bad practice.

Reply:
There's no reason to do this async unless your dialog is shown in the first place